src/main/java/org/mule/service/http/impl/service/server/grizzly/ResponseStreamingCompletionHandler.java

```diff
@@ -88,6 +90,8 @@ public void completed(WriteResult result) {
         // So, we have to call it manually here
         if (isDone && !httpResponsePacket.isChunked()) {
           doComplete();
+        } else if (isDone && httpResponsePacket.getRequest().getMethod().equals(HEAD)) {
```

Comment:
Same as 3.x. If the body will be ignored, perhaps we should avoid reacting to it in the first place.